### PR TITLE
Fix output results may  incorrect  when using intersect and except statements

### PR DESCRIPTION
**Pull request by yangzhg**
fix the bug desctibed in #3227 
add class SetOperationNode and move common funtions from intersect and except to it

### CODE REVIEW

File: be/src/exec/except_node.cpp [hunk 1]
```diff
@@ -179,17 +137,21 @@ Status ExceptNode::get_next(RuntimeState* state, RowBatch* out_batch, bool* eos)
...
+            out_batch->resize_and_allocate_tuple_buffer(state, &tuple_buf_size, &tuple_buf));
     uint32_t previous_hash = -1;
     TupleRow* previous_row = nullptr;
     while (_hash_tbl_iterator.has_next()) {
+        VLOG_ROW << "find row: "
+                 << get_row_output_string(_hash_tbl_iterator.get_row(), child(0)->row_desc())
+                 << " matched: " << _hash_tbl_iterator.matched();
         if (!_hash_tbl_iterator.matched()) {
             if (previous_hash != _hash_tbl_iterator.get_hash() ||
                 !equals(previous_row, _hash_tbl_iterator.get_row())) {
```
Comment (wutiangan) [inline be/src/exec/except_node.cpp]: why does hashtable has the same value？When you create a hash table, will duplicate values are inserted into the hash table？

File: be/src/exec/except_node.cpp [hunk 2]
```diff
@@ -112,6 +63,8 @@ Status ExceptNode::open(RuntimeState* state) {
         RETURN_IF_LIMIT_EXCEEDED(state, " Except, while constructing the hash table.");
         // build hash table and remvoe duplicate items
```
Comment (morningman) [inline be/src/exec/except_node.cpp]:
```suggestion
        // build hash table and remove duplicate items
```

File: be/src/exec/except_node.cpp (hunk 1 above)
Comment (morningman) [reply be/src/exec/except_node.cpp]: As I know, this hash table will store all rows with same hash key.

Comment (wutiangan) [reply be/src/exec/except_node.cpp]: But this is a set operation（It is different from join）. The set operation has deduplicated, so the elements of the hash table do not need to have duplicatited value。

Comment (yangzhg) [reply be/src/exec/except_node.cpp]: duplicated values will be iterate when get next,and deduplicate can reduce the cost of probe,you can find detail in hash_table.h,hash table in doris is not same with the common sense hashtable

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**